test/classes/forwarding/forwarding-sync-issue.chpl

```diff
@@ -0,0 +1,17 @@
+record ForwardingWrapper {
+  type eltType;
+
+  inline proc _value {
+    return new C(eltType);
+  }
+
+  forwarding _value;
+}
+
+
+class C {
+  type eltType;
+  var lock$ : sync bool;
+}
+
+var wrapper : ForwardingWrapper(int);
```

Comment:
I do not see what this test is testing. Add a comment about that?

Also, how about some action with wrapper? Like writeln(wrapper.lock$.isFull) ?